src/poetry/console/commands/source/add.py

```diff
@@ -50,8 +69,30 @@ def handle(self) -> int:
             )
             return 1
 
+        if is_default or is_secondary:
+            if priority is not None:
+                self.line_error(
+                    "Priority was passed through both --priority and a deprecated flag"
+                    " (--default or --secondary). Please only provide one of these."
+                )
+                return 1
+            else:
+                self.line(
+                    "Priority was set through both a deprecated flag (--default or"
+                    " --secondary). Consider using --priority next time."
+                )
```

Comment:
```suggestion
                self.line_error(
                    "<warning>Warning: Priority was set through a deprecated flag (--default or"
                    " --secondary). Consider using --priority next time.</warning>"
                )
```
(yellow color plus explicit word "Warning" for colorless IOs)